trino/integration_test.go

```diff
@@ -19,46 +19,103 @@ import (
 	"database/sql"
 	"errors"
 	"flag"
+	"log"
 	"net/http"
 	"os"
 	"strings"
 	"testing"
 	"time"
+
+	dt "github.com/ory/dockertest/v3"
 )
 
 var (
+	pool     *dt.Pool
+	resource *dt.Resource
+
 	integrationServerFlag = flag.String(
 		"trino_server_dsn",
 		os.Getenv("TRINO_SERVER_DSN"),
-		"dsn of the Trino server used for integration tests; default disabled",
+		"dsn of the Trino server used for integration tests instead of starting a Docker container",
 	)
 	integrationServerQueryTimeout = flag.Duration(
 		"trino_query_timeout",
 		5*time.Second,
 		"max duration for Trino queries to run before giving up",
 	)
+	noCleanup = flag.Bool(
+		"no_cleanup",
+		false,
+		"do not delete containers on exit",
+	)
 )
 
-func init() {
-	// explicitly init testing module so flags are registered before call to flag.Parse
-	testing.Init()
+func TestMain(m *testing.M) {
 	flag.Parse()
 	DefaultQueryTimeout = *integrationServerQueryTimeout
 	DefaultCancelQueryTimeout = *integrationServerQueryTimeout
-}
 
-// integrationServerDSN returns the URL of the integration test server.
-func integrationServerDSN(t *testing.T) string {
-	if dsn := *integrationServerFlag; dsn != "" {
-		return dsn
+	var err error
+	if *integrationServerFlag == "" && !testing.Short() {
+		pool, err = dt.NewPool("")
+		pool.MaxWait = 5 * time.Minute
```

Comment:
move below `err` check?

Also does it correspond to 20s timeout for trino being spinned up from the bash version? If so maybe make it a bit shorter wdyt?

Reply:
Fixed, thanks! I lowered this to 1 minute. In the bash script, the timeout didn't account for the `trino` command to be blocking. Here it's tracked independently.